nimble/host/include/host/ble_gatt.h

```diff
@@ -476,8 +477,15 @@ int ble_gatts_notify_custom(uint16_t conn_handle, uint16_t att_handle,
 /**
  * Deprecated. Should not be used. Use ble_gatts_notify_custom instead.
  */
+#if MYNEWT_VAL(BLE_SUPPRESS_DEPRECATE_WARN)
```

Comment:
I'd rather have this as an OPT-IN    MYNEWT_VAL(BLE_WARN_DEPRECATED)
and also define makro (eg in ble_hs.h) for this and not ifdef every single function declaration

#if MYNEWT_VAL(BLE_WARN_DEPRECATED)
#define NIMBLE_DEPRECATED(_msg) __attribute__((deprecated("Deprecated. Should not be used. " _msg))) 
#else
#define NIMBLE_DEPRECATED(_msg)
#endif

and than just do this to deprecate
NIMBLE_DEPRECATED("use bar") void foo(void);

Reply:
For this - NIMBLE_DEPRECATED("use bar") void foo(void);   CI always fails. Tried the other way around too.